projects/media-viewer/src/lib/shared/viewer.events.service.ts

```diff
@@ -0,0 +1,25 @@
+import {Injectable} from '@angular/core';
+import {Subject} from 'rxjs';
+
+
+// Viewer Custom-Event Types
+export interface TextSelectionRectangle {
+  x: number;
```

Comment:
this could get very large very quickly if we include the whole domain in here. might be worth splitting into separate files

Reply:
I agree, however @linusnorton suggested to place them where used. I’m happy to put these in a Types file again later on (in the same folder) if they grow too big

Reply:
I would suggest `annotations/annotation-set/text-selection-rectangle.model.ts`

Reply:
ok cool

Reply:
until i implement this a bit further post this merge, going to keep it in situ. As not settled on how to tap into the Rectangle implementation yet. For the moment, its purely a type to determine a TextSelection event

Reply:
@taleb @linusnorton are you both ok with this for now?